lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -37,7 +37,7 @@ namespace utf8
 
     inline bool ShouldFastPath(LPCUTF8 pb, LPCOLESTR pch)
     {
-        return (reinterpret_cast<size_t>(pb) & mAlignmentMask) == 0 || (reinterpret_cast<size_t>(pch) & mAlignmentMask) == 0;
+        return (reinterpret_cast<size_t>(pb) & mAlignmentMask) == 0 && (reinterpret_cast<size_t>(pch) & mAlignmentMask) == 0;
```

Comment:
Still don't quite understand how/when this fast path is taken but it's not blocking for this PR. The correctness hasn't changed with your change, so that should be ok. We should separately look at the effectiveness of ShouldFastPath...